packages/api/src/graphql/mutation.public.ts

```diff
@@ -99,6 +103,21 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
           }
         })
       }
+    },
+
+    updateComment: {
+      type: GraphQLNonNull(GraphQLPublicComment),
+      args: {
+        input: {type: GraphQLNonNull(GraphQLPublicCommentUpdateInput)}
+      },
+      async resolve(_, {input}, {dbAdapter}) {
```

Comment:
You need to check here if the user is authenticated. Also you should check if the user is the same as the comment they are trying to updated. Otherwise return a 403.

Reply:
@bishoymelek also as discussed check if the comment is in the `requestForChange` state.